operator/src/lib.rs

```diff
@@ -75,37 +76,142 @@ impl NifiState {
         mandatory_labels
     }
 
-    async fn create_config_map(&self, name: &str, config: &NifiConfig) -> Result<(), Error> {
+    async fn create_config_map(
+        &self,
+        cm_name: &str,
+        config: &NifiConfig,
+        node_name: &str,
+    ) -> Result<(), Error> {
+        let mut zk_ref: stackable_zookeeper_crd::util::ZookeeperReference =
+            self.context.resource.spec.zookeeper_reference.clone();
+
+        if let Some(chroot) = zk_ref.chroot.as_deref() {
+            stackable_zookeeper_crd::util::is_valid_zookeeper_path(chroot)?;
+        }
+
+        // retrieve zookeeper connect string
+        // we have to remove the chroot to only get the url and port
+        // nifi has its own config properties for the chroot and fails if the
+        // connect string is passed like: zookeeper_node:2181/nifi
+        zk_ref.chroot = None;
+
+        let zookeeper_info =
+            stackable_zookeeper_crd::util::get_zk_connection_info(&self.context.client, &zk_ref)
+                .await?;
+
+        debug!(
+            "Received zookeeper connect string: [{}]",
+            &zookeeper_info.connection_string
+        );
+
+        let bootstrap_conf = create_bootstrap_conf();
+        let nifi_properties = create_nifi_properties(
+            &self.context.resource.spec,
+            config,
+            &zookeeper_info.connection_string,
+            node_name,
+        );
+        let state_management_xml = create_state_management_xml(
+            &self.context.resource.spec,
+            &zookeeper_info.connection_string,
+        );
+
+        let mut data = BTreeMap::new();
+        data.insert("bootstrap.conf".to_string(), bootstrap_conf);
+        data.insert("nifi.properties".to_string(), nifi_properties);
+        data.insert("state-management.xml".to_string(), state_management_xml);
+
         match self
             .context
             .client
-            .get::<ConfigMap>(name, Some(&"default".to_string()))
+            .get::<ConfigMap>(cm_name, Some(&"default".to_string()))
             .await
         {
-            // TODO: check and compare content here
-            Ok(_) => {
-                debug!("ConfigMap [{}] already exists, skipping creation!", name);
-                return Ok(());
+            Ok(config_map) => {
+                if let Some(existing_config_map_data) = config_map.data {
+                    if existing_config_map_data == data {
+                        debug!(
+                            "ConfigMap [{}] already exists with identical data, skipping creation!",
+                            cm_name
+                        );
+                        return Ok(());
+                    } else {
+                        debug!(
+                            "ConfigMap [{}] already exists, but differs, recreating it!",
+                            cm_name
+                        );
+                    }
+                }
```

Comment:
This is okay but feels like a method for operator-rs

`compare_and_update` or something like that

Reply:
Yeah we have alot of code since the label selectors were introduced that we need to get into the operator-rs